### ISSUE TITLE
what is the detection_eval after 20k iterations?

### ISSUE THREAD

**Issue by XhqGlorry11** [amoussawi/caffe#5]
@amoussawi Hi, thanks for your work. You mentioned that the miss rate is 11.8% after 20k iterations. 
1. And I'd like to know what is the detection_eval that the network gives out on test set after 20k iterations?
2. I get the detecion_eval = 0.545697 after 8k iterations with fine-tuning. I think it's a little bit low. Is it normal or maybe something is wrong? What's your opinion?
Thanks.

**Comment by amoussawi** [comment]
1. The detection_eval was around 0.56 (in case of 512x512 image). 
2. The detection_eval you are getting is totally fine. 8k iterations are not enough. As you might have realized in the code, we were training the model for 10K iterations with 0.001 learning rate. 5k with 0.0001 LR. and another 5k with 1e-5 LR. After 10K iterations the loss doesn't decrease much, and we realized that using lesser LR after 10k iterations makes the loss more stable with less fluctuations, and it also makes the model converge and perform better.

**Comment by XhqGlorry11** [comment]
@

**Comment by XhqGlorry11** [comment]
@amoussawi Hi,

**Comment by XhqGlorry11** [comment]
@amoussawi Hi, I have trained the model for 20k iterations and the final detection_eval is around 0.558. And I run the sript "examples/ssd/ssd_caltech_video_ft.py", which calls for "build/tools/caffe", to evaluate the result. But when I run this script, the detection is really bad(shown in picutre 1). Then I used "build/examples/ssd/ssd_detect" to test the model and the input picture is screenshots of the video used before and get resonable result(shown in picture 2 and 3). What do you think may be the reason? 
Thx~
![1](https://cloud.githubusercontent.com/assets/22702027/26386291/6519c4e8-4078-11e7-9738-04f320210911.png)
![2](https://cloud.githubusercontent.com/assets/22702027/26386293/6569135e-4078-11e7-8c3e-73d6531161a7.png)
![3](https://cloud.githubusercontent.com/assets/22702027/26386292/6563ae64-4078-11e7-9a0b-88ff44c101a1.png)

**Comment by XhqGlorry11** [comment]
@amoussawi Hi, I have trained the model for 20k iterations and the final detection_eval is around 0.558. And I run the sript "examples/ssd/ssd_caltech_video_ft.py", which calls for "build/tools/caffe", to evaluate the result. But when I run this script, the detection is really bad(shown in picutre 1). Then I used "build/examples/ssd/ssd_detect" to test the model and the input picture is screenshots of the video used before and get resonable result(shown in picture 2 and 3). I watched the test video at: https://youtu.be/iKIW5Q0XAcg and the result is reasonable and I noticed that you were just running the script that I used before. What do you think may be the reason?
Thx~
![1](https://cloud.githubusercontent.com/assets/22702027/26386313/89b3f72e-4078-11e7-9cbe-a8dbdf99c652.png)
![2](https://cloud.githubusercontent.com/assets/22702027/26386312/89b0734c-4078-11e7-9a34-243bbb0afda2.png)
![3](https://cloud.githubusercontent.com/assets/22702027/26386326/9d5ea01c-4078-11e7-942d-adba45ad6b53.png)

**Comment by amoussawi** [comment]
Maybe you are using an old version of the script which had a small error. do a `git pull` and rerun. or just download the updated script directly from github repo, let me know if it's fixed. btw I'm interested in knowing what are you using this  pedestrian detection project for, of course if you don't mind telling me.

**Comment by XhqGlorry11** [comment]
@amoussawi Hi, I have downloaded the latest scripts from github and tried again just now. But the detection result is still as bad as I mentioned before. Can you try to use "examples/ssd/ssd_caltech_video_ft.py" to test your own model and see if it works fine? Just like the demo that you gave at https://youtu.be/iKIW5Q0XAcg.
Thx~

**Comment by amoussawi** [comment]
I'm sure the new script is correct. it was tested before being uploaded. what you are getting happens if the script is using the incorrect caffe model, or you have changed something in the hyperparameters and you didn't update the video script. Check the log if it's finetuning from the correct caffe model, and check if you have changed something in the hyperparameters. and also check the `loc_postfix` and `conf_postfix` if they are the same in the training script and the video testing script.

**Comment by XhqGlorry11** [comment]
@amoussawi OK, then there must be some mistakes when I am running the script. The training python script was modified a week ago and I forgot what modifications I have made. I will look through every details later.
Btw, I am trying to transform a pedestrian detection algorithm into an embedded system whose processor is Nvidia Tx1/Tx2 to create some mobile utilities. ^_^

**Comment by XhqGlorry11** [comment]
@amoussawi Hi, do you know how can I obtain the different test sets of caltech dataset? Such as reasonable, far, medium, etc. Are these test sets provided online? If so, where to get them?
Thx.
![1](https://cloud.githubusercontent.com/assets/22702027/26620031/aea91122-4613-11e7-9e67-acad2ce89a5e.png)

**Comment by amoussawi** [comment]
These are not test datasets. these are evaluation settings. You have to run `dbEval.m` in order to get these evaluations, assuming you have the caltech test dataset.

**Comment by XhqGlorry11** [comment]
@amoussawi Hi, it's ME AGAIN!! I am using this method to train another model on my own dataset and it's a 2 classes task(background and clock). But after I finished preparing data and start to train, the loss and mbox_loss are always 0. The annotations and label_map file are arranged as pic shows. And I ran create_list.sh and create_data.sh accordingly. But the loss stays 0 when training. Have you run into the same quesion and what do you think of it? 
Thx!!!

![screenshot from 2017-06-22 10 40 10](https://user-images.githubusercontent.com/22702027/27415121-d66eae80-5737-11e7-8c06-9d29df4e373a.png)

**Comment by amoussawi** [comment]
Hi. This project is only for caltech pedestrian detection. You should use the main SSD source code and cite their work only. Check for this issue in the main SSD repository.
